// app/database/db_manager.js
const mysql = require('mysql');
const setup = require('./config');

const connection = mysql.createConnection({
	host: setup.DB_HOST,
	port: setup.DB_PORT,
	user: setup.DB_USER,
	password: setup.DB_PWD,
	database: "mydb"
});

connection.connect((err) => {
	if (err) throw err;
	console.log("connected");
});

const perform = function(query, callback) {

	connection.query(query, (err, res, fields) => {
		if (err) throw err;
		callback(res);
	});
}

exports.currentTime = function(callback) {
	connection.query("SELECT DATE_ADD(NOW(), INTERVAL 1 HOUR)", (err, res) => {
		if (err) throw err;

		callback({
			time: res,
			success: true
		});
	});
}

exports.addRecord = function(tableName, fields, values, callback) {

	perform("INSERT INTO " + tableName + " " + fields + " VALUES " + values, (res) => {
		
		callback({
			data: res,
			success: true
		});
	});

}

exports.getRecords = function(tableName, callback) {

	perform("SELECT * FROM " + tableName, (res) => {
		
		callback({
			data: res,
			success: true
		});
	});
}

exports.deleteRecord = function(tableName, query, callback) {

	perform("DELETE FROM " + tableName + " WHERE " + query, (res) => {

		callback({
			data: res,
			success: true
		});
	});
}

//create TABLE within mydb database based on connection criteria
exports.createTable = function(tableName, tableFields, callback) {

	perform("CREATE TABLE " + tableName + " " + tableFields, (res) => {
		
		callback({
			data: res,
			success: true
		});
	});

}
